src/Datastore/RAMDataStore.cpp

```diff
@@ -66,21 +66,17 @@ TResult<User> RAMDataStore::removeUser(TSessionID const &ID) {
   }
 }
 
-// remove expired sessions
-TResultOpt RAMDataStore::checkSessionExpirations() {
+// check expired sessions
+TResult<bool> RAMDataStore::isSessionExpired(TSessionID const &ID) {
+  auto retUser = getUser(ID);
+  if (holds_alternative<Error>(retUser))
+    return get<Error>(retUser);
+
```

Comment:
can i do this here before acquiring the `mUserMutex`?

Problem:
i want to re-use the function `getUser` here. 
`getUser` acquires the same mutex `mUserMutex` as well. This would lead to a deadlock.

Reply:
maybe this will help:
http://www.cplusplus.com/reference/mutex/recursive_mutex/